extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/RuntimeBeanBuildItem.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.arc.deployment;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import org.jboss.builder.item.MultiBuildItem;
+
+/**
+ * Represents a bean that can be easily produced through a template (or other runtime Supplier implementation)
+ */
+public final class RuntimeBeanBuildItem extends MultiBuildItem {
+
+    final String scope;
+    //TODO: qualifiers
+    final String type;
+    final Supplier<Object> supplier;
+    final Set<String> qualifiers;
+
+    public RuntimeBeanBuildItem(String scope, String type, Supplier<Object> supplier, String... qualifiers) {
```

Comment:
Hm, this would deserve a builder class, with `Class<? extends Annotation>` or `io.quarkus.arc.processor.ScopeInfo` for scope, `AnnotationInstance...` for qualifiers, etc.